tests/test_echo.py

```diff
@@ -204,3 +187,32 @@ async def gen():
 
         assert [response.data for response in await async_iterable_to_list(
                 stub.EchoLastV2(gen()))] == [data * 20]
+
+
+@async_test
+async def test_purerpc_client_disconnect(echo_pb2, echo_grpc):
+    # when the client disconnects, the server should not log an exception
+    #
+    # NOTE: This test demonstrates a client/server test without multiprocessing or
+    #  fixture acrobatics.
+
+    # server
+    Servicer = make_servicer(echo_pb2, echo_grpc)
+    server = purerpc.Server(port=0)
+    server.add_service(Servicer().service)
+    async with anyio.create_task_group() as tg:
+        port = await tg.start(server.serve_async)
+
+        # client
+        async with purerpc.insecure_channel("localhost", port) as channel:
+            stub = echo_grpc.EchoStub(channel)
+
+            data = 'hello'
+            assert (await stub.Echo(echo_pb2.EchoRequest(data=data))).data == data
+
+            # close the sending stream, inducing EndOfStream on the server
+            await channel._grpc_socket._socket._stream.aclose()
```

Comment:
On Trio, it's resulting in `ClosedResourceError: another task closed this fd`.  I'm pretty lost.